db/version_set.h

```diff
@@ -858,6 +858,10 @@ class VersionSet {
                         std::vector<std::string>* manifest_filenames,
                         uint64_t min_pending_output);
 
+  bool IsGrabbedForPurge(uint64_t file_number) const;
+  void MarkAsGrabbedForPurge(uint64_t file_number);
+  void MarkAsPurged(uint64_t file_number);
```

Comment:
I was thinking about how we can verify the map is properly cleaned up. One idea is assert the map is empty in `DBImpl::CloseHelper` after `pending_purge_obsolete_files_` reaches zero. Would it work?

Reply:
DONE.